### ISSUE TITLE
potential workaround to make the workflow work with forks

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [Levi-Lesches/blocking-issues#19]
> @Levi-Lesches thankyou so much
> 
> i have a question about the use of the workflow. Are the following usecases valid for this workflow?
> 
>     1. PR1 is dependent on PR2 (PR2 has to be merged/closed first in order to get PR1 merged). This one is pretty straight forward.
> 
>     2. PR1 is dependent on issue1 (issue1 has to be closed first in order to get PR1 merged). This can be helpful when someone already created PR1 but u want issue1 to get addressed first but nobody has created a PR for it yet.
> 
>     3. issue1 is dependent on PR1 (PR1 has to be merged/closed first in order to close issue1). This can be helpful when somebody forgets to link PR1 as dependency when they create PR2 for issue1
> 
>     4. issue1 is dependent on issue2 (issue2 has to be closed first in order to close issue1). This can be helpful when there are no PR's yet that addresses one of the issues.

At the moment i cant get your workflow to trigger when i create a PR in the main repo from a fork. There is a workaround for this but i need your help with this. If u want the full details why workflows dont play nicely with forks read my discussion thread with another dev over here https://github.com/danilo-delbusso/pr-review-labeller/issues/1

I got one workflow to work with the workaround and now im trying to apply that same method to your workflow but without any luck. So it would be nice to have another set of eyes look at this issue because i believe it can be solved by using the workaround.

If u want to check out a workflow that does work this way u can use my ```conflicts.yml```
```
name: "Conflicts"
on:
  # So that PRs touching the same files as the push are updated
  push:
  # So that the `dirtyLabel` is removed if conflicts are resolve
  # We recommend `pull_request_target` so that github secrets are available.
  # In `pull_request` we wouldn't be able to change labels of fork PRs
  pull_request_target:
    types: [synchronize]
  workflow_run:
    workflows: ['Dummy workflow for conflicts']
    types: [requested]

jobs:
  main:
    runs-on: ubuntu-latest
    steps:
      - name: check if prs are dirty
        uses: eps1lon/actions-label-merge-conflict@releases/2.x
        with:
          dirtyLabel: "PR: merge conflicts / rebase needed"
          removeOnDirtyLabel: "PR: waiting for review"
          repoToken: "${{ secrets.GITHUB_TOKEN }}"
          commentOnDirty: "This pull request has conflicts, please resolve those before we can evaluate the pull request."
          commentOnClean: "Conflicts have been resolved. A maintainer will review the pull request shortly."
```
dummy workflow ```dummy-conflicts.yml```
```
name: Dummy workflow for conflicts
on:
  pull_request_review:
    types: [submitted]
jobs:
  dummy:
    runs-on: ubuntu-latest
    steps:
      - run: echo "this is a dummy workflow that triggers a workflow_run; it's necessary because otherwise the repo secrets will not be in scope for externally forked pull requests"
```

I highly recommend reading the full thread i linked because it will get u a better understanding why this workaround is needed.

**Comment by Levi-Lesches** [comment]
Interesting, the [logs](https://github.com/Levi-Lesches/blocking-issues-test/runs/7682868709?check_suite_focus=true) from closing https://github.com/Levi-Lesches/blocking-issues-test/pull/39 say: 
```
Getting current issue...
Issue is closed. Unblocking other issues...
Unblocking issues blocked by #39...
Initializing labels...
Updating 40
      Re-running most recent action on testpr
```
Oh... I see why this is failing. To signal to GitHub that the PR is ready to be merged, it needs to re-run the check that failed and let it succeed. When an issue/PR is closed, it goes through all issues/PRs that were blocked by it and, if it's a PR, re-runs the failed action.

The problem here is _how_ it finds the action. Currently, the code searches for all checks that have run on a given branch, with the logic being that one of them has to be `blocking-issues`. However, the branch in question (`testpr`) has no workflows because it's a fork with actions disabled. 

So I need to rework the logic around re-running failed checks. Maybe we don't even need to re-run it and can just start a new check.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
This workflow does workflow work for me now without any weird workaround https://github.com/eps1lon/actions-label-merge-conflict maybe u can take a look at it

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Update: @Levi-Lesches I found a similar workflow to this that does work from fork maybe u can look into that https://github.com/gregsdennis/dependencies-action
